app/models/week.rb

```diff
@@ -2,4 +2,13 @@ class Week < ActiveRecord::Base
   belongs_to :edition
 
   scope :order_by_number, -> { order(number: :asc) }
+
+  def self.get_previous_week_by_date(date)
+    week_for_date = Edition.current.weeks.find_by(start: date.beginning_of_week)
+    Edition.current.weeks.find_by(number: (week_for_date.number - 1))
```

Comment:
why not simply do `Edition.current.weeks.find_by(start: 1.week.ago.beginning_of_week`)?

Reply:
sure. Should have thought of that. 👍